jsapp/js/account/stripe.api.ts

```diff
@@ -199,6 +199,37 @@ const getRecurringAddOnLimits = (limits: AccountLimit) => {
   return newLimits;
 };
 
+/**
+ * Add one-time addon limits to already calculated account limits
+ */
+const addOneTimeAddOnLimits = (
+  limits: AccountLimit,
+  oneTimeAddOns: OneTimeAddOn[]
+) => {
+  oneTimeAddOns.forEach((addon) => {
+    if (
+      addon.limits_remaining.submission_limit &&
+      limits.submission_limit !== Limits.unlimited
+    ) {
+      limits.submission_limit += addon.limits_remaining.submission_limit;
+    }
+    if (
+      addon.limits_remaining.asr_seconds_limit &&
+      limits.nlp_seconds_limit !== Limits.unlimited
+    ) {
+      limits.nlp_seconds_limit += addon.limits_remaining.asr_seconds_limit;
+    }
+    if (
+      addon.limits_remaining.mt_characters_limit &&
+      limits.nlp_character_limit !== Limits.unlimited
+    ) {
+      limits.nlp_character_limit +=
+        addon.limits_remaining.mt_characters_limit;
+    }
+  });
```

Comment:
This bit should take into account `addon.is_available` - if it's false, that specific add-on shouldn't be added to the limit.

It's not really documented in the code yet, so no worries on missing it, but if a customer no longer has the subscription that the add-on requires, or they refund the underlying Charge/PaymentIntent, the add-on will become unavailable. Add-ons can become available again if the customer upgrades back to the necessary subscription.